### DISCUSSION TITLE
24KHZ/24KHZ MP3

### DISCUSSION THREAD

**Discussion by zinzer** [enzo1982/freac#496] [General Discussion]
I am just using this program for the first time. I wish to set both the bit rate and the sample frequency to 24KHZ . I see the bitrate slider, but how do I set the sample frequency?
Thanks,

**Comment by enzo1982** [comment]
Hi, to change the sample frequency, please enable/use the _Sample Rate Converter_ component on the _Processing_ page of the configuration dialog.

**Comment by zinzer** [comment]
[enzo1982]  Thanks for your reply. I have followed your instructions and set the sample rate to 24KHZ, but mediainfo shows that the converted file still has a sample rate of 16KHZ. Am I missing something?

**Reply by enzo1982** [reply]
Ah, yes, LAME also performs its own sample frequency conversion to improve quality if the selected bit rate is too low for the input frequency.

You can disable that conversion by checking the _Disable all filtering_ box on the _Audio processing_ tab of the LAME configuration. But be warned, quality of 24 kbps/24 kHz will be much worse than 24 kbps/16 kHz. In a quick test I could immediately hear strong encoding artifacts in the 24 kHz version and I'm not very good at hearing artifacts normally...

**Comment by zinzer** [comment]
Robert.  Thank you for your reply, your advice, and this useful software. I am re-encoding audio books. Some of these have sample rates of 320KHZ stereo, and so unnecessary large  file sizes in GHZ. 
I have used 24KHZ/24KHZ before without noticing harmonics, but I will do some experiments.
Maybe all the MP3 controls could be on a single page in a later version of fre:ac?

Thanks again

Terry (Glastonbury UK)

**Reply by enzo1982** [reply]
> Maybe all the MP3 controls could be on a single page in a later version of fre:ac?

The options from all four tabs on one page would be a bit overwhelming I think. And I'd really like to keep the expert settings (and that includes the _Audio processing_ tab) separate as changing them can easily cause degraded output quality.

However, I'm now thinking about having only two tabs, _Basic_ and _Advanced_ with the first two existing tabs in the first one and the expert options in the second.

**Reply by mrfragger** [reply]
> Robert. Thank you for your reply, your advice, and this useful software. I am re-encoding audio books. Some of these have sample rates of 320KHZ stereo, and so unnecessary large file sizes in GHZ.

I did thousands of chaptered audiobooks m4b 32kbps 22khz AAC HE_v2.  I did that to save some space over 44 or 48Khz.

Now since the last two years I've done all opus chaptered audiobooks 16kbps 48khz using fre:ac. BTW all opus audio is 48Khz.  Remember it's just the frequency of pitch of voice.  Even transcription I do it only accepts wav files at 16Khz. Opus is half the file size as m4b 32kbps.  For example 60MB audiobook is 8 hours. 914MB audiobook is 123 hours. In my listening tests it's just as good as AAC 32kbps.  Music it distorts it obviously but that might be just a short intro on some podcast for instance and no big deal.

Fre-ac you can encode to a single file and video players let you navigate to the chapters.  If chapters with different sample rates, bitrates, etc. simply re-encode them all to 16 or 32kbps opus then encode them to a single file otherwise chapter times will be off. I put a black png image for covers and use the following players so all audiobooks have subtitles using whisper.cpp
Linux SMPlayer, Mac IINA, Windows PotPlayer, iOS Nplayer $5, Android mpv-android.  

To name chapters to their filename use python titlecase so Lord Of The Rings ---> Lord of the Rings.  If you put chapter number make sure you put a : (colon) after so it captilalizes after chapter number Chapter 1: To Be Or Not To Be --> Chapter 1: To Be or Not to Be  ...if you don't then Chapter 1 To Be Or Not To Be --> Chapter 1 to Be or Not to Be
`for f in *.opus ; do mv -i -- "$f" "$(printf '%s\n' "$f" | titlecase)" ; kid3-cli -c "set title '$(printf '%n' "$f")'" "$f" ; done`

This is the blackcover I use with a aspect ratio of 2:35:1 and only 1KB. 
![blackcover](https://github.com/enzo1982/freac/assets/80211672/eb944a01-c768-461c-b85e-1f663f0dea4e)

If you ever need to do some metdata editing use python puddletag.


![smplayer](https://github.com/enzo1982/freac/assets/80211672/5412d689-f52f-484a-98de-7e32be2f1554)
This is showing opus audiobook with subs on SMPlayer on Linux.

![nplayer](https://github.com/enzo1982/freac/assets/80211672/1cefd9df-50d7-402f-b8a6-d6d0239c723c)
This is showing opus audiobook with subs on nPlayer on iOS.

**Comment by zinzer** [comment]
mrfragger
Thanks for all the information and wisdom.

I saw an audiobook enced in flac the other day, -crazy!
